crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -142,8 +144,41 @@ impl RoomList {
         let stream = stream! {
             loop {
                 let filter_fn = filter_fn_cell.take().await;
-                let (values, stream) = list
-                    .room_list_stream()
+                let (raw_values, mut raw_stream) = list.room_list_stream();
```

Comment:
I see you're getting the `room_list_stream()` to maintain `raw_values` (= the list entries) over time. In my opinion it should be just fine to call `list.room_list()` around line 166, so you don't have to maintain the raw list over changes we're receiving (that should simplify the select branch for the raw stream nicely).

Reply:
That would be a bug! and a bug doesn't exist until a test case proves it does :sunglasses: :eyes: `</hint>`

Reply:
In crates/matrix-sdk-ui/tests/integration/room_list_service.rs I now show that the order is relevant by emitting an update before calling stream.next() and after calling stream.next (Variation 1, Variation 2)